presto-main/src/main/java/io/prestosql/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -331,8 +332,20 @@ private void rebuildHashAggregationBuilder()
                 operatorContext,
                 Optional.of(DataSize.succinctBytes(0)),
                 joinCompiler,
-                false,
-                false);
+                () -> {
+                    updateMemory();
+                    // TODO: Support GroupByHash yielding in spillable hash aggregation.
+                    // This is particularly important as revocable memory reservations are
+                    // large and GroupByHash rehashing (without yielding) might put memory
+                    // pressure on GC.
+                    // In order to support GroupByHash yielding here we should:
+                    // 1. Remove dependency on InMemoryHashAggregationBuilder.
+                    //    SpillableHashAggregationBuilder should use GroupByHash directly.
+                    // 2. GroupByHash should have methods that don't return work, but
+                    //    return next page position (addPage method) or partial GroupByIdBlock
+                    //    (getGroupIds method) when input Page was not fully consumed.
```

Comment:
Maybe move the above plan to an issue and leave only `// TODO: Support GroupByHash yielding in spillable hash aggregation (https://...issue/123)` here?